### PR TITLE
Remove dependency on saelens from pyproject, add to demo.ipynb

### PR DESCRIPTION
**Pull request by hijohnnylin**
Addresses https://github.com/callummcdougall/sae_vis/issues/42

@chanind please check if you have a min. thanks!

### CODE REVIEW

**Comment by hijohnnylin** [comment]
Hmm pyright failed in github workflow but when I run it locally it succeeds. Do I need to do something different?

```
~/sae_vis - [move_saelens_dep] $ poetry run pyright

added 2 packages, and audited 3 packages in 1s

found 0 vulnerabilities
0 errors, 0 warnings, 0 informations 
```

File: pyproject.toml [hunk 1]
```diff
@@ -13,10 +13,10 @@ einops = "^0.7.0"
 datasets = "^2.0.0"
 dataclasses-json = "^0.6.4"
 jaxtyping = "^0.2.28"
-sae-lens = "^0.5.0"
 transformer-lens = "^1.0.0"
 eindex-callum = "^0.1.0"
 rich = "^13.7.1"
+matplotlib = "^3.8.4"
```
Comment (chanind) [inline pyproject.toml:19]: why is matplotlib needed?

**Comment by chanind** [comment]
👍 lgtm

File: pyproject.toml (hunk 1 above)
Comment (hijohnnylin) [reply pyproject.toml:19]: previously pyright failed with:
```
Run poetry run pyright

added 1 package, and audited 2 packages in 2s

found 0 vulnerabilities
/home/runner/work/sae_vis/sae_vis/sae_vis/html_fns.py
  /home/runner/work/sae_vis/sae_vis/sae_vis/html_fns.py:6:6 - error: Import "matplotlib" could not be resolved (reportMissingImports)
1 error, 0 warnings, 0 informations 
Error: Process completed with exit code 1.
```

Comment (chanind) [reply pyproject.toml:19]: Ah I guess that dependency was previously satisfied by depending on SAELens. Makes sense for that to be a direct dependency if it's used in the project

**Pull request merged**